easybuild/tools/module_generator.py

```diff
@@ -238,18 +285,181 @@ def set_alias(self, key, value):
-        _log.debug("Updating fake for this ModuleGenerator instance to %s (was %s)" % (fake, self.fake))
-        self.fake = fake
-        # fake mode: set installpath to temporary dir
-        if self.fake:
-            self.tmpdir = tempfile.mkdtemp()
-            _log.debug("Fake mode: using %s (instead of %s)" % (self.tmpdir, self.module_path))
-            self.module_path = self.tmpdir
+
+class ModuleGeneratorLua(ModuleGenerator):
+    """
+    Class for generating Lua module files.
+    """
+    SYNTAX = LUA_SYNTAX
+
+    MODULE_SUFFIX = '.lua'
+
+    def __init__(self, *args, **kwargs):
+        """ModuleGeneratorLua constructor."""
+        super(ModuleGeneratorLua, self).__init__(*args, **kwargs)
+
+    def module_header(self):
+        """Return module header string."""
+        return ''
+
+    def comment(self, msg):
+        """Return string containing given message as a comment."""
+        return " -- %s\n" % msg
+
+    def get_description(self, conflict=True):
+        """
+        Generate a description.
+        """
+
+        description = "%s - Homepage: %s" % (self.app.cfg['description'], self.app.cfg['homepage'])
+
+        lines = [
+            "local pkg = {}",
+            "help = [["
+            "%(description)s"
+            "]]",
+            "whatis([[Name: %(name)s]])",
+            "whatis([[Version: %(version)s]])",
+            "whatis([[Description: %(description)s]])",
+            "whatis([[Homepage: %(homepage)s]])"
+            "whatis([[License: N/A ]])",
+            "whatis([[Keywords: Not set]])",
+            "",
+            "",
+            'pkg.root="%(installdir)s"',
+            "",
+            ]
+
+        #@todo check if this is really needed, imho Lmod doesnt need this at all.
+        if self.app.cfg['moduleloadnoconflict']:
+            lines.extend([
+             'if ( not isloaded("%(name)s/%(version)s")) then',
+             '  load("%(name)s/%(version)s")',
+             'end',
+             ])
+
+        elif conflict:
+            # conflicts are not needed in lua module files, as Lmod's one name
+            # rule and automatic swapping.
+            pass
+
+        txt = '\n'.join(lines) % {
+            'name': self.app.name,
+            'version': self.app.version,
+            'description': description,
+            'installdir': self.app.installdir,
+            'homepage': self.app.cfg['homepage'],
+        }
+
+        return txt
+
+    def load_module(self, mod_name):
+        """
+        Generate load statements for module.
+        """
+        if build_option('recursive_mod_unload'):
+            # not wrapping the 'module load' with an is-loaded guard ensures recursive unloading;
+            # when "module unload" is called on the module in which the depedency "module load" is present,
+            # it will get translated to "module unload"
+            load_statement = ['load("%(mod_name)s")']
```

Comment:
if this is the same load string as in the else block, use a varaible for it to ensure it is the same